integration-libs/opf/payment/core/connectors/opf-payment.adapter.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  AfterRedirectScriptResponse,
+  OpfPaymentVerificationPayload,
+  OpfPaymentVerificationResponse,
+  PaymentInitiationConfig,
+  PaymentSessionData,
+  SubmitCompleteRequest,
+  SubmitCompleteResponse,
+  SubmitRequest,
+  SubmitResponse,
+} from '@spartacus/opf/payment/root';
+import { Observable } from 'rxjs';
+
+export abstract class OpfPaymentAdapter {
+  /**
+   * Abstract method used to verify payment
+   */
+
+  abstract verifyPayment(
+    paymentSessionId: string,
+    payload: OpfPaymentVerificationPayload
+  ): Observable<OpfPaymentVerificationResponse>;
+
+  /**
+   * Abstract method used to submit payment for hosted-fields pattern
+   */
+
+  abstract submitPayment(
+    submitRequest: SubmitRequest,
+    otpKey: string,
+    paymentSessionId: string
+  ): Observable<SubmitResponse>;
+
+  /**
+   * Abstract method used to submit-complete payment for hosted-fields pattern
+   */
+
+  abstract submitCompletePayment(
+    submitRequest: SubmitCompleteRequest,
+    otpKey: string,
+    paymentSessionId: string
+  ): Observable<SubmitCompleteResponse>;
+
+  /**
+   * Abstract method used to get AfterRedirect scripts used in hosted-fields pattern
+   */
+  abstract afterRedirectScripts(
```

Comment:
[MEDIUM]

For consistency and readability, let's start the methods names with a verb. In this case `get` seems appropriate, e.g. `getAfterRedirectScripts`.

Note: an analogical comment applies to many places in this PR with the method `afterRedirectScripts`.